lib/ecl/ecl_smspec.cpp

```diff
@@ -108,15 +108,18 @@ struct ecl_smspec_struct {
     smspec_node instances. The actual smspec_node instances are
     owned by the smspec_nodes vector;
   */
+  node_map             field_var_index;
+  node_map             misc_var_index;            /* Variables like 'TCPU' and 'NEWTON'. */
+  node_map             gen_var_index              /* This is "everything" - things can either be found as gen_var("WWCT:OP_X") or as well_var("WWCT" , "OP_X") */;
+
   std::map<std::string, node_map> well_var_index; /* Indexes for all well variables:
                                                      {well1: {var1: index1 , var2: index2} , well2: {var1: index1 , var2: index2}} */
-  std::map<std::string, std::map<int, node_map>> well_completion_var_index; /* Indexes for completion indexes .*/
-  hash_type          * group_var_index;            /* Indexes for group variables.*/
-  hash_type          * field_var_index;
-  hash_type          * region_var_index;           /* The stored index is an offset. */
-  hash_type          * misc_var_index;             /* Variables like 'TCPU' and 'NEWTON'. */
-  hash_type          * block_var_index;            /* Block variables like BPR */
-  node_map             gen_var_index              /* This is "everything" - things can either be found as gen_var("WWCT:OP_X") or as well_var("WWCT" , "OP_X") */;
+  std::map<std::string, node_map> group_var_index; /* Indexes for group variables.*/
+  std::map<std::string, std::map<int, smspec_node_type*>> region_var_index;  /* The stored index is an offset. */
+  std::map<std::string, std::map<int, smspec_node_type*>>  block_var_index;            /* Block variables like BPR */
+  std::map<std::string, std::map<int, node_map>>           well_completion_var_index; /* Indexes for completion indexes .*/
```

Comment:
The `region_var_index`, `block_var_index` and `well_completion_var_index` members should be changed to:
```
std::map<int, node_map>
```
that way the final lookup can be based on `node_map` for all variables.